src/python/grpcio/grpc/_cython/_cygrpc/aio/call.pxd.pxi

```diff
@@ -28,4 +28,15 @@ cdef class _AioCall(GrpcCallWrapper):
         # because Core is holding a pointer for the callback handler.
         bint _is_locally_cancelled
 
+        # Following attributes are used for storing the status of the call and
+        # the initial metadata. Waiters are used for pausing the execution of
+        # tasks that are asking for one of the field when they are not yet
+        # available.
+        object _status
+        object _initial_metadata
+        list _waiters_status
+        list _waiters_initial_metadata
```

Comment:
Is there a reason to use multiple waiters (`asyncio.Future`) than just one? By reading the implementation, you could create one universal Future object, and have all invocations `await` on it. I wonder if reducing the number of pending futures could help performance.

Reply:
Yeps the reason is one of the comments that made in the PR

> This PR also addresses an issue with the previous code where isolated tasks that were asking for the initial metadata or the RPC status might leave polluted the path with an asyncio.CasncelledError exception. With the changes proposed multiple calls can be done to the initial_metadata and status coroutines provided by AioCall without polluting the other waiters.

There is a new test implemented here [1] which if you try to run it in master will fail.

There is a recurrent thought in my head. if we all agree that the usage of multiple waiters is the way to go for isolating concurrent access to the coros that are informing about the status or other attributes of an ongoing RPC, should be the same rationale appliable to the `await call`?

[1] https://github.com/grpc/grpc/pull/21696/files#diff-ebe6c75bf1347f8032019b99fbf4a8bdR133

Reply:
Make sense! As for `await call`, it depends on our definition. If users want to cancel a Task that is pending on an RPC, do they want to cancel the RPC or just the waiter? I think we could merge the two ideas: we use multiple waiters, but if all existing waiters have been cancelled, then cancel the underlying RPC. WDYT?

Also, I benchmarked `asyncio.Event`, it is slower than Future:
```Python
completed_future = loop.create_future()
completed_future.set_result(None)

async def await_completed_future():
    await completed_future

completed_event = asyncio.Event()
completed_event.set()

async def await_completed_event():
    await completed_event.wait()
```
```
Spent 0.23s for 1000000 runs of await_completed_future
233.17 ns
Spent 0.43s for 1000000 runs of await_completed_event
434.45 ns
```

Reply:
Having the feeling that any change regarding the `await __call__` would need to be handled in another PR - if everybody agrees that the current one already provides some value and the `await __call__` is a different problem (that could be solved with the same pattern). 

Also if we want to tackle again the `await __call__` most likely is gonna impact again - sorry for that - with how cancellation is being managed. So, I would prefer to have all of the arguments and a good story compressed in a new tiquet. And having all of us aligned with the goal of this.

Reply:
I'm open to improvement ideas. Yes, it is not in the scope of this PR.